File: lib/app/home/recording/recorded_duration_tracker.dart
class RecordedDurationTracker {
  DateTime? _wallClockSessionStart;
  DateTime? _activeSegmentStart;
  Duration _accumulated = Duration.zero;

  bool get hasStarted => _wallClockSessionStart != null;
  bool get isPaused => hasStarted && _activeSegmentStart == null;

  void start([DateTime? now]) {
    final startedAt = now ?? DateTime.now();
    _wallClockSessionStart = startedAt;
    _activeSegmentStart = startedAt;
    _accumulated = Duration.zero;
  }

  void pause([DateTime? now]) {
    final activeSegmentStart = _activeSegmentStart;
    if (activeSegmentStart == null) return;
    final pausedAt = now ?? DateTime.now();
    _accumulated += pausedAt.difference(activeSegmentStart);
    _activeSegmentStart = null;
  }

  void resume([DateTime? now]) {
    if (_wallClockSessionStart == null || _activeSegmentStart != null) return;
    _activeSegmentStart = now ?? DateTime.now();
  }

  Duration current([DateTime? now]) {
    final activeSegmentStart = _activeSegmentStart;
    if (activeSegmentStart == null) return _accumulated;
    return _accumulated +
        (now ?? DateTime.now()).difference(activeSegmentStart);
  }

  void reset() {
    _wallClockSessionStart = null;
    _activeSegmentStart = null;
    _accumulated = Duration.zero;
  }
}

File: macos/Runner/Capture/Indicator/RecordedDurationTracker.swift
import Foundation

/// Tracks wall-clock recording time across pause/resume segments.
/// Pure value type — no platform dependency (engine-domain; see windows-port-inventory §7).
struct RecordedDurationTracker {
  private(set) var wallClockSessionStart: Date?
  private var activeSegmentStart: Date?
  private(set) var accumulatedRecordedDuration: TimeInterval = 0
  private(set) var isPaused = false

  mutating func start(at date: Date = Date()) {
    wallClockSessionStart = date
    activeSegmentStart = date
    accumulatedRecordedDuration = 0
    isPaused = false
  }

  mutating func pause(at date: Date = Date()) {
    guard let activeSegmentStart else { return }
    accumulatedRecordedDuration += max(0, date.timeIntervalSince(activeSegmentStart))
    self.activeSegmentStart = nil
    isPaused = true
  }

  mutating func resume(at date: Date = Date()) {
    guard wallClockSessionStart != nil, activeSegmentStart == nil else { return }
    activeSegmentStart = date
    isPaused = false
  }

  mutating func stop(at date: Date = Date()) {
    if let activeSegmentStart {
      accumulatedRecordedDuration += max(0, date.timeIntervalSince(activeSegmentStart))
    }
    activeSegmentStart = nil
    isPaused = false
  }

  mutating func reset() {
    wallClockSessionStart = nil
    activeSegmentStart = nil
    accumulatedRecordedDuration = 0
    isPaused = false
  }

  func currentRecordedDuration(at date: Date = Date()) -> TimeInterval {
    guard let activeSegmentStart else { return accumulatedRecordedDuration }
    return accumulatedRecordedDuration + max(0, date.timeIntervalSince(activeSegmentStart))
  }
}
